binaries/coordinator/src/lib.rs

```diff
@@ -703,6 +705,45 @@ async fn start_inner(
                 tracing::info!("Daemon `{daemon_id}` exited");
                 daemon_connections.remove(&daemon_id);
             }
+            Event::DataflowSpawnResult {
+                dataflow_id,
+                daemon_id,
+                result,
+            } => match running_dataflows.get_mut(&dataflow_id) {
+                Some(dataflow) => {
+                    dataflow.pending_spawn_results.remove(&daemon_id);
+                    match result {
+                        Ok(()) => {
+                            if dataflow.pending_spawn_results.is_empty() {
+                                tracing::info!("successfully spawned dataflow `{dataflow_id}`");
+                                if let Some(reply_tx) = dataflow.spawn_result_tx.take() {
+                                    let _ =
+                                        reply_tx.send(Ok(ControlRequestReply::DataflowStarted {
+                                            uuid: dataflow_id,
+                                        }));
+                                }
+                            }
+                        }
+                        Err(err) => {
+                            tracing::warn!("error while spawning dataflow `{dataflow_id}`");
+                            if let Some(reply_tx) = dataflow.spawn_result_tx.take() {
+                                let _ = reply_tx.send(Err(err));
+                            }
+                        }
+                    };
+                }
+                None => {
+                    tracing::warn!("received DataflowSpawnResult, but no matching dataflow in `running_dataflows` map");
+                }
+            },
+        }
+
+        let elapsed = start.elapsed();
+        if elapsed > Duration::from_millis(100) {
+            tracing::warn!(
+                "Coordinator took {}ms for handling event: {event_debug}",
+                elapsed.as_millis()
+            );
         }
```

Comment:
Seems like this is debug code no?

Reply:
Event handling in the main loop of the coordinator should be quick, otherwise handling of subsequent events will be delayed. This warning indicates that we block the event loop too long, which is something that we should fix.

So I think it might be a good idea to keep the warning, as it makes it easier to notice some potential performance issues in the future. Or what do you think?